app/controllers/api/connect/v4/systems/products_controller.rb

```diff
@@ -1,5 +1,6 @@
 class Api::Connect::V4::Systems::ProductsController < Api::Connect::V3::Systems::ProductsController
 
+  after_action :refresh_system_token, only: %i[synchronize destroy], if: -> { request.headers.key?(SYSTEM_TOKEN_HEADER) }
```

Comment:
```suggestion
  after_action :refresh_system_token, only: %i[activate upgrade synchronize destroy], if: -> { request.headers.key?(SYSTEM_TOKEN_HEADER) }
```

Reply:
v4 product controller has only two actions synchronize and destroy and has after_action to refresh for these two actions 
v3 product controller has only activate and upgrade actions. and has after_action to refresh for these two actions 
V4 inherits v3
so I thought, we don't need to add it in v4.

Reply:
One would think so, but it seems the `after_action` definition in the V4 class overwrites the one from V3 and needs to include `activate upgrade`. I tested this with the test steps from the pr description, and it fails without this change.

Reply:
Got it. Added activate and upgrade actions in the v4 product controller